README.md

```diff
@@ -28,4 +28,263 @@ One of the very common use cases of WebXR is 180 and/or 360 photos or videos, bo
+
+```js
+// Render Loop for a projection layer with a WebGL framebuffer source.
+const xrGfx = new XRWebGLGraphicsBinding(xrSession, gl);
+const layer = xrGfx.requestProjectionLayer();
+
+xrSession.updateRenderState({ layers: [layer] });
+xrSession.requestAnimationFrame(onXRFrame);
+
+function onXRFrame(time, xrFrame) {
+  xrSession.requestAnimationFrame(onXRFrame);
+
+  gl.bindFramebuffer(gl.FRAMEBUFFER, xrGfx.getCurrentFramebuffer(layer));
+  for (let view in xrViewerPose.views) {
+    let viewport = layer.getViewSubImage(view).viewport;
+    gl.viewport(viewport.x, viewport.y, viewport.width, viewport.height);
+    
+    // Render from the viewpoint of xrView
+  }
+}
+```
+
+In some cases, such as a mono `XRQuadLayer` being shown on a stereo device, multiple `XRView`s may return the same `XRSubImage` values. To avoid rendering the same view multiple times in these scenarios the `primary` attribute of the `XRSubImage` will be be set to `false` for all but one of the overlapping views. (It should be noted that `XRProjectionLayer`s will never return `XRSubImage`s with `primary` set to false.)
+
+```js
+// Render Loop for a projection layer with a WebGL framebuffer source.
+const xrGfx = new XRWebGLGraphicsBinding(xrSession, gl);
+
+const quadLayer = xrGfx.requestQuadLayer({ pixelWidth: 512, pixelHeight: 512, stereo: false });
+// Position 2 meters away from the origin with a width and height of 1.5 meters
+quadLayer.referenceSpace = xrReferenceSpace;
+quadLayer.transform = new XRRigidTransform({z: -2});
+quadLayer.width = 1.5;
+quadLayer.height = 1.5;
+
+xrSession.updateRenderState({ layers: [quadLayer] });
+xrSession.requestAnimationFrame(onXRFrame);
+
+function onXRFrame(time, xrFrame) {
+  xrSession.requestAnimationFrame(onXRFrame);
+
+  gl.bindFramebuffer(gl.FRAMEBUFFER, xrGfx.getCurrentFramebuffer(quadLayer));
+  for (let view in xrViewerPose.views) {
+    let subImage = quadLayer.getViewSubImage(view);
+
+    if (subImage.primary) {
+      gl.viewport(viewport.x, viewport.y, viewport.width, viewport.height);
+      
+      // Render content for view.eye
+    }
+  }
+}
+```
+
+## Appendix A: Proposed IDL
+
+```webidl
+//
+// Layer interface
+//
+
+interface XRLayer {
+  XRSubImage? getViewSubImage(XRView view);
+
+  readonly attribute unsigned long pixelWidth;
+  readonly attribute unsigned long pixelHeight;
+  readonly attribute unsigned long arraySize;
+  readonly attribute boolean ignoreDepthValues;
+
+  boolean blendTextureSourceAlpha = false;
+  boolean chromaticAberrationCorrection = false;
+}
+
+interface XRSubImage {
+  readonly attribute XRViewport viewport;
+  readonly attribute unsigned long imageIndex;
+  readonly attribute boolean primary;
+}
+
+//
+// Layer types
+//
+
+interface XRProjectionLayer extends XRLayer {
+}
+
+interface XRQuadLayer extends XRLayer {
+  attribute XRRigidTransform transform;
```

Comment:
doesn't have to be resolved here but I suggested during the F2F that we just use offset spaces, reducing the number of places where we're doing matrix math. We might want to file a followup once this lands.